tools/esp-api/espservice.cpp

```diff
@@ -25,45 +25,51 @@ using namespace std;
 
 int EspService::sendRequest()
 {
-
     int responseCode = -1;
     StringBuffer req,res,status;
-    req.append(formArgs);
+    req.append(reqString);
 
     Owned<IHttpClientContext> httpctx = getHttpClientContext();
     Owned <IHttpClient> httpclient = httpctx->createHttpClient(NULL, url);
 
-    if(!streq(username, "") && !streq(password, ""))
+    if(!isEmptyString(username))
     {
         httpclient->setUserID(username);
+    }
+    if(!isEmptyString(password))
```

Comment:
There is no use case where a password is added without a username that I am aware of. This condition should only be executed when a username is given.